### ISSUE TITLE
Incorrect group template in libopenldap.sh, creates group under ou=users!

### ISSUE THREAD

**Issue by jamshid** [bitnami/bitnami-docker-openldap#2]
**Description**

Using https://hub.docker.com/r/bitnami/openldap when I set `LDAP_GROUP=mygroup` the startup script `/opt/bitnami/scripts/libopenldap.sh` creates this strange group entry:

```
DN: ou=mygroup,ou=users,dc=example,dc=org
	cn: mygroup

	objectClass: groupOfNames

	member: cn=myuser,ou=users,dc=example,dc=org

	ou: mygroup
```

I'd expect it to be under `ou=group,dc=example,dc=org` and I'd expect the `dn` to be `cn=mygroup,...` not `ou=mygroup`.

**Steps to reproduce the issue:**

1. 
```
 docker run -p 1389:1389 --detach --rm --name openldap \
  --env LDAP_ADMIN_USERNAME=admin \
  --env LDAP_ADMIN_PASSWORD=adminpassword \
  --env LDAP_GROUP=mygroup \
  --env LDAP_USERS=myuser \
  --env LDAP_PASSWORDS=mypassword \
  bitnami/openldap:latest
```
2. 
```
$ curl 'ldap://localhost:1389/dc=example,dc=org?*?sub?(objectclass=*)'
...
DN: ou=mygroup,ou=users,dc=example,dc=org
	cn: mygroup

	objectClass: groupOfNames

	member: cn=myuser,ou=users,dc=example,dc=org

	ou: mygroup
```

**Describe the results you received:**
Group created as:
`DN: ou=mygroup,ou=users,dc=example,dc=org`
**Describe the results you expected:**
Group should be:
`DN: cn=mygroup,ou=group,dc=example,dc=org`

**Additional information you deem important (e.g. issue happens only occasionally):**

Was hard to find this would be nice if you provided a link back here at https://hub.docker.com/r/bitnami/openldap

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
```

- Output of `docker info`:

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
 Kubernetes:
  Version:          v1.16.6-beta.0
  StackAPI:         v1beta2
jambook:dockerfiles jamshid$ docker info
Client:
 Debug Mode: false

Server:
 Containers: 74
  Running: 60
  Paused: 0
  Stopped: 14
 Images: 930
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 11.71GiB
 Name: docker-desktop
 ID: C5K5:PHIG:2JTC:W2JU:4KEJ:YC4V:XCT3:M74C:426J:OYP3:MZLL:AVM5
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 442
  Goroutines: 385
  System Time: 2020-07-02T02:16:48.852867947Z
  EventsListeners: 3
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  192.168.1.61:5000
  jambook.local:5000
  jambook.local:5555
  localhost:5000
  127.0.0.1:5000
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine```

- Output of `docker-compose version` (if applicable):

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Docker for Mac

**Comment by andresbono** [comment]
Hi @jamshid, thanks for filling this issue. We will try to reproduce it and we will get back to you.

**Comment by andresbono** [comment]
Hi again @jamshid, it might have more sense to finish the hierarchy of the tree with a `cn` in the group creation.

In the result that you expected, you are also indicating `ou=group` instead of `ou=users`. In this case, we just wanted to indicate that the organizational unit is all about users.

In any case, you might need to edit this basic sample data with your own schema.

Just to let you know, we plan to improve this container to allow mounting your own `ldif` files that will be loaded during the first initialization and executed with `ldapadd` to populate the data in the way you want. But I can't tell you when this will be addressed.

**Comment by andresbono** [comment]
Please check #3 and let us know what you think!

**Comment by jamshid** [comment]
Thank you for the fix so that it now creates the group with a correct `dn` like:
```
DN: cn=mygroup,ou=users,dc=example,dc=org
	cn: mygroup
	objectClass: groupOfNames
	member: cn=myuser,ou=users,dc=example,dc=org
```
But I still think it should create the group in an `ou=groups` not `ou=users`:
```
DN: cn=mygroup,ou=groups,dc=example,dc=org
	cn: mygroup
	objectClass: groupOfNames
	member: cn=myuser,ou=users,dc=example,dc=org
```
Groups should be under a different `ou`, not `users` or `people` right? E.g.
https://serverfault.com/questions/87079/openldap-posix-groups-account
```
# User primary group
dn: cn=ussergroup,ou=groups,dc=me,dc=com
...
# User account
dn: uid=user,ou=users,dc=me,dc=com
```
So maybe add a `LDAP_GROUP_OU` similar to `LDAP_USER_DC` and use that in the template. 
Btw `LDAP_USER_DC` should probably be renamed to "LDAP_USER_OU", since it defaults to `ou=users` not a `dc`?

---
Thanks, the ability to add users and groups via an ldif would be great, though the current env vars are very convenient. 
FYI I'm trying to use this image as a simple default LDAP server for a demo environment, ideally without having to create a custom image. But I'd also like to replace an old custom openldap image that has a bunch of test users and groups.

Maybe allow the ldif or simpler list of users and passwords filename to be specified by an env variable for configuration at startup time. K8s or docker-compose could make this file available to the container at run time as a "secrets" file, keeping the passwords out of env vars and configuration files. Maybe I should try it and submit a patch.

**Comment by andresbono** [comment]
> So maybe add a `LDAP_GROUP_OU` similar to `LDAP_USER_DC` and use that in the template.
> Btw `LDAP_USER_DC` should probably be renamed to "LDAP_USER_OU", since it defaults to `ou=users` not a `dc`?

It makes sense @jamshid. Would you like to contribute with a new PR with these changes?
 
> Maybe allow the ldif or simpler list of users and passwords filename to be specified by an env variable for configuration at startup time. K8s or docker-compose could make this file available to the container at run time as a "secrets" file, keeping the passwords out of env vars and configuration files. Maybe I should try it and submit a patch.

Having a PR from you, even if it is a draft, would be great!

**Comment by andresbono** [comment]
Hi, we have merged #3, those are really minor changes. Please open a new PR if you want to add more features!
